e2e_tests/tests/test_sdk.py

```diff
@@ -474,3 +474,32 @@ def test_rp_workspace_mapping() -> None:
     finally:
         for wid in workspace_ids:
             bindings.delete_DeleteWorkspace(session=sess, id=wid)
+
+
+@pytest.mark.e2e_cpu
+def test_experiment_w_template() -> None:
+    sess = api_utils.user_session()
+    detobj = client.Determined._from_session(sess)
+    # create template
+    tpl = bindings.v1Template(
+        name=api_utils.get_random_string(),
```

Comment:
It's not a big deal, but I think it's nice and simpler to keep randomness out of functions when that randomness isn't explicitly used for something. In this case this means that deciding if the randomness tests something you want to test and:
* if yes, creating a separate test testing that behavior explicitly
* if no, replacing it with a constant name